pipeline_manager/src/main.rs

```diff
@@ -509,13 +651,21 @@ struct CancelProjectRequest {
 ///
 /// The client should poll the `/project_status` endpoint
 /// to determine when the cancelation request completes.
-///
-/// # HTTP errors
-///
-/// * `NOT_FOUND` - `project_id` does not exist in the database.
-///
-/// * `CONFLICT` - project version specified in the request doesn't match the
-///   latest project version in the database.
+#[utoipa::path(
+    request_body = CancelProjectRequest,
+    responses(
+        (status = ACCEPTED, description = "Cancelation request submitted."),
+        (status = NOT_FOUND
+            , description = "Specified `project_id` does not exist in the database."
+            , body = ErrorResponse
+            , example = json!(ErrorResponse::new("Unknown project id '42'"))),
+        (status = CONFLICT
+            , description = "Project version specified in the request doesn't match the latest project version in the database."
+            , body = ErrorResponse
+            , example = json!(ErrorResponse::new("Outdated project version '{3}'"))),
+    ),
+    tag = "Project"
+)]
 #[post("/cancel_project")]
 async fn cancel_project(
```

Comment:
What's the difference between canceling a project and deleting it?

Reply:
`/cancel` cancels compilation project compilation.  The project remains in the database and can be compiled/modified/queried.

Reply:
To follow a create/<modify>/delete cycle for a project, here is an approach that comes to mind.

* `POST /projects/`     # creates a project, not yet compiled
* `GET /projects/{project_id}`
* `POST /compiler/{project_id}` # creates a compilation job
* `GET /compiler/{project_id}` # gets the status of the compilation job
* `DELETE /compiler/{project_id}` # ends the compilation job

An alternative that some people use, which might be overkill for now:

* `POST /projects/`     # creates a project, not yet compiled
* `GET /projects/{project_id}`
* `PUT/PATCH /projects/{project_id}`   # modify a "compilation" field in the project that changes its desired state, which triggers compilation. AWS APIs have this approach to attaching "Action" sub-resources to resources.